tests/h/schemas/base_test.py

```diff
@@ -87,6 +116,90 @@ def test_it_returns_all_errors_in_message(self):
         assert message.startswith("'foo' is a required property, 'bar' is a required property")
 
 
+class TestRemoveUnknownProperties(object):
+
+    @pytest.mark.parametrize('data,expected', [
+        (
+            # data
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string"
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": "string",
+             "baz": {
+                "foo": "string",
+                "no_exist": "string",
+                "bar": {
+                    "foo": "string",
+                    "no_exist": "string",
+                }
+             }},
+            # expected
+            {"foo": "string",
+             "baz": {
+                "foo": "string",
+                "bar": {
+                    "foo": "string",
+                }
+             }},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "no_exist": {
+                "bar": "string",
+             }},
+            # expected
+            {"foo": "string"},
+
+        ),
+        (
+            # data
+            {"foo": "string",
+             "bar": {
```

Comment:
"bar" is defined as being an integer in the schema. `remove_unused_properties` should never be called with this input because the data should have been validated beforehand and this input would fail validation.

Additionally I found it surprising that `bar` is expected to be stripped from the validated data since `bar` is a valid field in the schema.

I discovered this when trying to figure out if the `remove_unused_properties` implementation could be simplified.

Reply:
See https://github.com/hypothesis/h/pull/5224/files#r212778211.